Makefile

```diff
@@ -29,15 +34,25 @@ init:
 	@echo "+++ Making $@ +++"
 	make -f Makefile.work $@
 
-clean configure reset showtag sonic-slave-build sonic-slave-bash :
+#
+# Function to invoke target $@ in Makefile.work with proper BLDENV
+#
+define make_work
 	@echo "+++ Making $@ +++"
-ifeq ($(NOJESSIE), 0)
-	make -f Makefile.work $@
-endif
-ifeq ($(NOSTRETCH), 0)
-	BLDENV=stretch make -f Makefile.work $@
-endif
+	$(if $(NOJESSIE),, make -f Makefile.work $@)
+	$(if $(NOSTRETCH),, BLDENV=stretch make -f Makefile.work $@)
```

Comment:
`$(if $(NOSTRETCH),,BLDENV=stretch make -f Makefile.work $@)` always evaluated to be `true` and nothing will be invoked here. 

Suggest to change it to:
```
-ifeq ($(NOJESSIE),0)
-BUILD_JESSIE=1
-endif
-
-ifeq ($(NOSTRETCH),0)
-BUILD_STRETCH=1
-endif
-
 PLATFORM_PATH := platform/$(if $(PLATFORM),$(PLATFORM),$(CONFIGURED_PLATFORM))
 PLATFORM_CHECKOUT := platform/checkout
 PLATFORM_CHECKOUT_FILE := $(PLATFORM_CHECKOUT)/$(PLATFORM).ini
@@ -47,8 +39,8 @@ init:
 #
 define make_work
        @echo "+++ Making $@ +++"
-       $(if $(BUILD_JESSIE), make -f Makefile.work $@,)
-       $(if $(BUILD_STRETCH),BLDENV=stretch make -f Makefile.work $@,)
+       $(if $(NOJESSIE),, make -f Makefile.work $@)
+       $(if $(NOSTRETCH),, BLDENV=stretch make -f Makefile.work $@)
        BLDENV=buster make -f Makefile.work $@
 endef
 ```